File: Checkpoint Manager/CheckpointManager/course.cpp
#include "course.h"
#include <stdexcept>

//The courses class represents a model of a course

Course::Course(std::string id)
{
    this->id = id;
}


void Course::addCheckpoint(Node* checkpoint){
    this->checkpoints.push_back(checkpoint);
}

//Get checkpoint number X
Node* Course::getCheckpointNo(int no){
    try {
        return this->checkpoints.at(no);
    }catch (const std::out_of_range& oor) {
        return NULL;
    }
}

std::string Course::getId(){
    return this->id;
}

File: Event Creator/Event Creator/src/uk/ac/aber/awf1/Course.java
/*
 * Course class
 * 
 * This class represents a course
 * 
 */
package uk.ac.aber.awf1;

/**
 *
 * @author fewstera
 */
class Course {
    
    char id;
    
    Course(char id){
        this.id = id;
    }
    
    public char getId(){
        return this.id;
    }
    
}
